File: ExtApp.Model/Sys/Dept.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExtApp.Model
{
    /// <summary>
    /// Organizational Chart
    /// </summary>
    public class Dept
    {
        /// <summary>
        /// Organization ID
        /// </summary>
        public virtual int ID { get; set; }

        /// <summary>
        /// Parent Dept
        /// </summary>
        public virtual Dept PDept { get; set; }

        /// <summary>
        /// Code
        /// </summary>
        public virtual string Code { get; set; }

        /// <summary>
        /// Name
        /// </summary>
        public virtual string Name { get; set; }

        /// <summary>
        /// Layer
        /// </summary>
        public virtual int Layer { get; set; }

        /// <summary>
        /// Status
        /// </summary>
        public virtual int Status { get; set; }

        /// <summary>
        /// Memo
        /// </summary>
        public virtual string Memo { get; set; }
    }
}

File: ExtApp.Web/app/model/sys/dept.js
﻿/**
* Organization model
*/

Ext.define('ExtApp.model.sys.Dept', {
    extend: 'Ext.data.Model',

    fields: [
        'ID', {
            name: 'PID',
            mapping: function (data) {
                return data.PDept == null ? 0 : data.PDept.ID;
            }
        }, {
            name: 'PName',
            mapping: function (data) {
                return data.PDept == null ? '' : data.PDept.Name;
            }
        },
        'Code',
        'Name',
        'Layer',
        'Status',
        'Memo'
    ]
});